lib/filterx/filterx-eval.h

```diff
@@ -138,7 +138,7 @@ static inline void
 filterx_eval_store_weak_ref(FilterXObject *object)
 {
   /* Preserved objects do not need weak refs. */
-  if (object && filterx_object_is_preserved(object))
+  if (object && (filterx_object_is_preserved(object) || filterx_object_is_readonly(object)))
```

Comment:
Is this correct?

What if I put the subobject of a readonly dict into a writable dict?